### PR TITLE
[IGNORE] update release docs

### PR DESCRIPTION
**Pull request by juliepagano**
I did my first release a while back and found myself getting a little confused with some of the directions and also recall a few instructions from other maintainers that weren't written down. This is an attempt to reword and expand a few bits to make it a little more straightforward for a new maintainer to cut a release. I'm very open to feedback on all the changes here!

# Checklist

- [x] Pull request has a descriptive title and context useful to a reviewer.
- [x] Pull request title follows the `[<catalog_entry>] <commit message>` naming convention using one of the following `catalog_entry` values: `FEATURE`, `ENHANCEMENT`, `BUGFIX`, `BREAKINGCHANGE`, `IGNORE`.
- [x] All commits have [DCO signoffs](https://github.com/probot/dco#how-it-works).

## UI Changes

- [x] Changes that impact the UI include screenshots and/or screencasts of the relevant changes.
- [x] Code follows the [UI guidelines](https://github.com/perses/perses/blob/main/ui/ui-guidelines.md).

### CODE REVIEW

File: RELEASE.md
Comment (juliepagano) [inline RELEASE.md:105]: @sjcobb I think you were the person who mentioned the recommendation to use "merge" (instead of "squash and merge"). Do you have any additional context that should get added here?

File: RELEASE.md
Comment (juliepagano) [inline RELEASE.md:90]: @cndonovan I recall you having issues with the GPG key bit of this. Is there anything you learned that's worth including here for future folks?

File: RELEASE.md
Comment (juliepagano) [inline RELEASE.md:76]: @Nexucis the mdox docs validator is complaining about this link because it's a duplicate. It seemed like maybe they fixed this behavior in https://github.com/bwplotka/mdox/pull/26/files, but I think we're on the latest version, so maybe I'm missing something. Any thoughts? If there isn't an obvious fix, I can remove the link to avoid the issue.

File: RELEASE.md
Comment (juliepagano) [inline RELEASE.md:97]: @Nexucis is this bit accurate? Anything notable I'm missing? Are the docker images/binaries published anywhere else?

File: RELEASE.md
Comment (Nexucis) [inline RELEASE.md:48]:
```suggestion
**/!\ Release candidates and patch releases**
```

File: RELEASE.md
Comment (Nexucis) [reply RELEASE.md:97]: the docker images are only published on docker hub:  https://hub.docker.com/r/persesdev/perses

They are not pushed to github if it's what you mean.

File: RELEASE.md
Comment (Nexucis) [reply RELEASE.md:105]: the branch `release vX.Y` is containing the commit tagged. So if you are going to squash this branch in `main`, then this commit will disappear in the git history and then the tag is not attached to any known commit.

That's why it's import to use `merge` instead because it will preserve the commits made on the release branch

File: RELEASE.md
Comment (Nexucis) [reply RELEASE.md:76]: so far I don't see an obvious fix excepting putting in the file `.mdox.validate.yaml` the link to ignore it. But that means it won't check anymore if the link is valid 😐

File: RELEASE.md
Comment (Nexucis) [inline RELEASE.md:80]: I'm wondering if the section title is still accurate. Perhaps `create the release tag` would be accurate

File: RELEASE.md
Comment (sjcobb) [reply RELEASE.md:105]: I found the original discussion on Matrix when this came up [here](https://matrix.to/#/!XNmgYIAndZOkEvQAbb:matrix.org/$jLLD7HCHNbdxMDZQJOHNqUt0ku5jFhD2QSkLfWTs1DE?via=matrix.org&via=herail.net). You can see what happened too if you check some older releases like [v0.7.1](https://github.com/perses/perses/tree/v0.7.1):

image

At one point, I looked through all the settings in GitHub to see if we could automate this somehow and couldn't find a way, it gets tricky too since it always defaults to however you merged your last PR so sometimes it's annoying to remember to change back to 'Squash' the next time since we do want to Squash every other type of PR 😕

Comment (sjcobb) [reply RELEASE.md:105]: One unknown for me is, does this only happen on patch releases? The example above is `v0.7.1` where I was merging in multiple commits that got squashed. I'm pretty sure before that release, I did squash some releases that were normal minor release versions and not a patch release, and it worked fine 🤷‍♂️

Comment (sjcobb) [reply RELEASE.md:105]: I'm wondering again about the whole minor release branch workflow, it definitely has some upsides about letting development continue on `main` branch and pushing out quick patch releases for bugfixes, but since we have to sync the minor release branch with main anyways before the next minor release can go out, that negates some of the benefits to me. 

This came up with Eunice's breaking changes in our last `v0.28.0` release. I [posted on Matrix](https://matrix.to/#/!XNmgYIAndZOkEvQAbb:matrix.org/$uy-7vPmyIjUtjlxncWV06smk453TS_C3pVo7iTbRLvo?via=matrix.org&via=herail.net) that I wanted to keep the `release/v0.27` open in case we needed to get bugfixes out before we could manually migrate all our dashboards internally (side note: we really need some automated way to do these schema migrations). But then when Eunice went to do the release, we realized that keeping that branch open wasn't an option since we needed to sync the changelog and version changes before she could do her release. 

I've posted about maybe changing to a workflow where we cherrypick just the PRs we need back into an older release branch to do patch releases w/o holding up `main` but the downside there is we might run into this issue where the tags end up in state like `v0.7.1` did (`This commit does not belong to any branch on this repository` error). Any thoughts @juliepagano are there other options we aren't considering? For context, we picked our current process because it works well for the Prometheus project ([ex](https://github.com/prometheus/prometheus/blob/main/RELEASE.md)), but since their releases go out every ~6 weeks and ours are more frequent, the needs are somewhat different

Comment (sjcobb) [reply RELEASE.md:105]: Related, I wrote up a concise version of the steps I did to do the v0.27.0 release [here](https://github.com/perses/perses/pull/1098):

image

Now that we have changelog generation that works really well and GitHub gives us the ability to manually edit release notes after the release goes out to clean up any PRs with 'unknowns', I bet we could automate at least half of the bullet points. What if we could just cut a release by making a `release/v0.x.x` branch and then basically everything automatically happens including the last steps where the release is tagged (assuming a maintainer cut the release branch and we figure out some way to do the GPG key signing in CI)

File: RELEASE.md
Comment (sjcobb) [reply RELEASE.md:90]: I had an issue when I first got my M1 Mac that I posted about [here](https://matrix.to/#/!XNmgYIAndZOkEvQAbb:matrix.org/$1RHQzJMV44vquFgoNTPaKm8ozFcOhR-XrpWEAXkAZAk?via=matrix.org&via=herail.net), at the time, this SO article helped me: https://stackoverflow.com/questions/39494631/gpg-failed-to-sign-the-data-fatal-failed-to-write-commit-object-git-2-10-0/40066889#40066889

I thought about adding a note about that in RELEASE.md back then but it did feel very specific to my setup and I was hoping the issue when get fixed eventually, I think Christine might have run into it too though, curious did you have any similar issues with your M1?

File: RELEASE.md
Comment (juliepagano) [reply RELEASE.md:97]: Gotcha! Looks like just the go binaries are in those tarballs. I knew I wasn't getting this quite right. Will reword this.

File: RELEASE.md
Comment (juliepagano) [reply RELEASE.md:76]: For this case, I'm just going to take out the link, but I created https://github.com/perses/perses/issues/1122 for us to look into this a little more deeply when we have time.

File: RELEASE.md
Comment (juliepagano) [reply RELEASE.md:80]: The steps include waiting on the automated generation of the release and validating it after. What about "Create tag and validate release"?

File: RELEASE.md
Comment (juliepagano) [reply RELEASE.md:105]: @sjcobb would you be up for moving the thoughts about adjusting the process to a github discussion? I think that will be a slightly better venue for us to work through tradeoffs and how we want to adjust things. I could see it taking a bit to work out something new, so I think there's value in having this PR focus on clearly defining our current process until we land on something new.

Comment (sjcobb) [reply RELEASE.md]: Yeah agreed! Thanks for updating this by the way 🙏

**Review by sjcobb — APPROVED** [review]

File: RELEASE.md
Comment (eunicorn) [inline RELEASE.md:52]:
```suggestion
- Create a branch using the release branch `release/v<X.Y>` you just created in the step above (e.g. `yourname/release-v<major>.<minor>.<patch>`).
```

I know you mentioned it in the previous step, but I feel like it's better to emphasize it here

File: RELEASE.md
Comment (eunicorn) [reply RELEASE.md:91]: i remember running into this issue as well

**Review by eunicorn — APPROVED** [review]

**Comment by juliepagano** [comment]
oh noooo, I always forget that allowing github suggestions makes the DCO angry...will fix and amend that commit.

File: RELEASE.md
Comment (juliepagano) [reply RELEASE.md:91]: I added a note about that issue and a link to the SO article.

File: RELEASE.md
Comment (juliepagano) [reply RELEASE.md]: Updated to "Create release tag and validate release."

File: RELEASE.md
Comment (cndonovan) [inline RELEASE.md:48]: I actually don't understand what a release candidate is here (this already existed, but since you're nicely updating all this, I figure I'll share what's confusing me too) 

The release branch is `release/v<major>.<minor>`. If we have to patch it, this is saying the process is to create a branch and patch it into the release branch (right?)

That leaves me wondering, what is a release candidate / how is different from a release branch?

Comment (cndonovan) [reply RELEASE.md:48]: Mainly because it says: `Do not create release/<version> for patch or release candidate releases.` Isn't the branch the release candidate?

File: RELEASE.md
Comment (cndonovan) [inline RELEASE.md:52]: Create a branch from (instead of using)?

Comment (cndonovan) [reply RELEASE.md:52]: I might say: "Create a branch from... [rest of sentence]. You might name it something like: `yourname/release-v<major>.<minor>.<patch>`"

The reason is I say this is that as I read this sentence, I got confused and thought the release branch was called `yourname/...`

File: RELEASE.md
Comment (cndonovan) [inline RELEASE.md:101]: not big deal, sometimes we're camel-casing GitHub and sometimes we're all lowercasing it (github).

**Review by cndonovan — APPROVED** [review]
thanks!

File: RELEASE.md
Comment (Nexucis) [reply RELEASE.md:48]: a release candidate is a release suffixed by `rc<number of the release candidate>`. For example `v0.28.0-rc0`
Making a release candidate or a release is the same process, it just change the tag itself. 

Let me know if it helps you to understand :) I hope it doesn't confuse you even more

**Pull request merged**
